### PR TITLE
Swap base constant product algorithm

### PR DESCRIPTION
**Pull request by akhilesh-singh-Lab49**
**Description**:

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: .openzeppelin/unknown-31337.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
+{
+  "manifestVersion": "3.2",
```
Comment (amandeeprehal-lab49) [inline .openzeppelin/unknown-31337.json:2]: pls remove this line.

File: contracts/AbstractSwap.sol [hunk 2]
```diff
@@ -124,4 +125,37 @@ abstract contract AbstractSwap is HederaResponseCodes {
...
+        int64 amountTokenA = ((invariantValue) / (pair.tokenB.tokenQty - amountTokenB)) - (pair.tokenA.tokenQty * precision);
+        return amountTokenA;
+    }
+
+    function getVariantValue() public view returns(int64) {
+        int64 precision = getPrecisionValue();
+        return (pair.tokenA.tokenQty * pair.tokenB.tokenQty) * precision;
+    }
+
+    function getPrecisionValue() internal pure returns(int64) {
```
Comment (amandeeprehal-lab49) [inline contracts/AbstractSwap.sol:158]: Pls  make it public

File: integrationTest/swapV2.ts [hunk 3]
```diff
@@ -178,12 +178,75 @@ const getContributorTokenShare = async () => {
...
+  const getSpotPrice = await new ContractExecuteTransaction()
+    .setContractId(contractId)
+    .setGas(1000000)
+    .setFunction("getSpotPrice")
+    .freezeWith(client);
+  const getPairQtyTx = await getSpotPrice.execute(client);
+  const response = await getPairQtyTx.getRecord(client);
+  const price = response.contractFunctionResult!.getInt256(0);
+
+  console.log(`spot price for token is ${price}. \n`);
```
Comment (amandeeprehal-lab49) [inline integrationTest/swapV2.ts:191]: pls change console log message  for A/B

File: test/swapv2-test.ts [hunk 4]
```diff
@@ -4,6 +4,9 @@ import {  expect } from "chai";
 import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
 import { ethers, upgrades } from "hardhat";
 
+import {BigNumber} from "bignumber.js";
```
Comment (amandeeprehal-lab49) [inline test/swapv2-test.ts:7]: pls remove it.

File: .openzeppelin/unknown-31337.json (hunk 1 above)
Comment (akhilesh-singh-Lab49) [reply .openzeppelin/unknown-31337.json]: removed file and added to gitignore.

File: contracts/AbstractSwap.sol (hunk 2 above)
Comment (akhilesh-singh-Lab49) [reply contracts/AbstractSwap.sol]: Done

File: integrationTest/swapV2.ts (hunk 3 above)
Comment (akhilesh-singh-Lab49) [reply integrationTest/swapV2.ts]: Done

File: test/swapv2-test.ts (hunk 4 above)
Comment (akhilesh-singh-Lab49) [reply test/swapv2-test.ts]: Done

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**